INTEGRATING.md

```diff
@@ -1,36 +1,171 @@
+After deploying this charm, we can relate `Grafana Agent` and `Zinc` through `loki_push_api` relation interface:
 
-- Any charm that supports the `prometheus_scrape` interface.
-You can integrate it using the Prometheus Charm library. The documentation of the charm library, available through [Charmhub](https://charmhub.io/prometheus-k8s/libraries) provides further details.
+```bash
+juju relate grafana-agent-k8s:logging-provider zinc-k8s
+```
+
+
+Once the relation is established, Zinc charm can start sending logs to Grafana Agent charm.
+
+### Grafana dashboard
+
+```yaml
+  grafana-dashboard:
+    interface: grafana_dashboard
+```
 
+Over the `grafana-dashboard` relation using the [`grafana-dashboard`](https://charmhub.io/grafana-k8s/libraries/grafana_dashboard) interface, this Grafana Agent charm also provides meaningful dashboards about its metrics to be shown in a [Grafana Charm ](https://charmhub.io/grafana-k8s).
 
-- Any charm that supports the `prometheus_remote_write` interface.
-You can integrate it using the Prometheus Charm library. The documentation of the charm library, available through [Charmhub](https://charmhub.io/prometheus-k8s/libraries) provides further details.
+In order to add these dashboards to Grafana all that is required is to relate the two charms in the following way:
 
+```bash
+juju relate \
+    grafana-agent-k8s:grafana-dashboard \
+    grafana-k8s:grafana-dashboard
+```
+
+### Self metrics endpoint
+
+
+```yaml
+self-metrics-endpoint:
+    interface: prometheus_scrape
+```
+This Grafana Agent charm may forward information about its metrics endpoint and associated alert rules to a Prometheus charm over the `self-metrics-endpoint` relation using the [`prometheus_scrape`](https://charmhub.io/prometheus-k8s/libraries/prometheus_scrape) interface. In order for these metrics to be aggregated by the remote Prometheus charm all that is required is to relate the two charms as in:
+
+```bash
+juju relate \
+    grafana-agent-k8s:self-metrics-endpoint \
+    prometheus-k8s:metrics-endpoint
+```
+
+## Requires
+
+### Logging consumer
+
+```yaml
+  logging-consumer:
+    interface: loki_push_api
+```
 
-Besides this charm also provides the `loki_push_api` interface through the `log_proxy`, so can be used as a Log Proxy between a workload charm and a charm that also implements the `loki_push_api` interface, for instance [Loki](https://charmhub.io/loki-k8s). The documentation of the charm library, available through [Charmhub](https://charmhub.io/loki-k8s) provides further details.
+[Loki](https://charmhub.io/loki-k8s) receives logs forwarded from Grafana Agent, aggregates and deduplicates them.
+Grafana Agent Charmed Operator relates to Loki over the `loki_push_api` interface using the `logging-consumer` relation.
+
+
+```bash
+juju relate grafana-agent-k8s:logging-consumer loki-k8s
+```
+
+
+
+### Metrics endpoint
+
+```yaml
+  metrics-endpoint:
+    interface: prometheus_scrape
+```
```

Comment:
I wonder if we should change the format a bit so things are more obvious and more copy-pasteable.
For example:

> ### Metrics endpoint
> Charms may forward information about their metrics endpoints and associated alert rules to the Grafana Agent charm over the `metrics-endpoint` relation using the [`prometheus_scrape`](https://charmhub.io/prometheus-k8s/libraries/prometheus_scrape) interface.
> 
> #### This charm
> ```yaml
> requires:
>   metrics-endpoint:
>     interface: prometheus_scrape
> ```
> 
> #### Remote charm
> 
> ```yaml
> provides:
>   metrics-endpoint:
>     interface: prometheus_scrape
> ```

etc.

Reply:
I like this format!